### PR TITLE
Use transcript in consensus chain tests

### PR DESCRIPTION
**Pull request by oxarbitrage**
- Transcript tests not using `VERIFY_TIMEOUT_SECONDS`
- `continuous_blockchain_test` not included.

Part of https://github.com/ZcashFoundation/zebra/issues/610

### CODE REVIEW

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]: Issue: Use names that are specific, and different from other names.

```suggestion
static BLOCK_VERIFY_TRANSCRIPT_GENESIS: Lazy<Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>> =
```

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]:
```suggestion
static BLOCK_VERIFY_TRANSCRIPT_GENESIS_FAIL: Lazy<Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>> =
```

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]:
```suggestion
static BLOCK_VERIFY_TRANSCRIPT_GENESIS_TO_BLOCK_1: Lazy<Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>> =
```

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]: Nit: The state service is not a verifier:

```suggestion
    /// SPANDOC: Make sure the state service is ready (1/2)
```

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]: Nit: This is actually a ChainVerifier service, not a BlockVerifier service.

So let's be consistent with the other functions:
```suggestion
    /// SPANDOC: Make sure the verifier service is ready (1/2)
```

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]:
```suggestion
    /// SPANDOC: Make sure the verifier service is ready (2/2)
```

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]:
```suggestion
    /// SPANDOC: Make sure the state service is ready (2/2)
```

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]:
```suggestion
static STATE_VERIFY_TRANSCRIPT_GENESIS: Lazy<
```

**Review by teor2345 — CHANGES_REQUESTED** [review]
Thanks for doing this work, it removes a lot of duplicate code.

I think transcripts are a good idea, but we have to be careful how we use them.

This change:
* makes some tests less readable
* removes test timeouts, which makes deadlocks harder to debug

Here's what we can do to improve these tests:
- [ ] use specific, unique names for each transcript, which describe what they do
- [ ] if the tests are still unclear, add more comments (let's check this after the rename)
- [ ] add a TimeoutLayer to the transcripts in these tests:
  - https://docs.rs/tower/0.3.1/tower/timeout/struct.TimeoutLayer.html

We're about to change the BlockVerifier and state service implementations, so we will need these timeouts to diagnose any new deadlocks.

Next time, please change one test per commit. That makes it easier to check that the transcripts are the same as the original tests.

**Comment by teor2345** [comment]
Blocked by #861, because we need transcript timeouts to replace tests which use timeouts.

**Comment by teor2345** [comment]
Let's use a TimeoutLayer to wrap the transcripts in these tests:
https://docs.rs/tower/0.3.1/tower/timeout/struct.TimeoutLayer.html

(See the discussion in #861.)

**Comment by oxarbitrage** [comment]
@teor2345 I added the timeout layer at https://github.com/ZcashFoundation/zebra/pull/858/commits/b7fc6818a9d64fc88616e655038639be9dc5cc64 however not totally sure if it is ok. Please take a look when you can, thanks.

File: zebra-consensus/src/chain/tests.rs
Comment (yaahc) [inline zebra-consensus/src/chain/tests.rs:117]: Is there a reason to keep these as statics? @ZcashFoundation/zebra-team 

I think the readability issues @teor2345 brought up might be helped by just inlining the content of these setup statics into the tests that use them where possible.

File: zebra-consensus/src/chain/tests.rs
Comment (yaahc) [inline zebra-consensus/src/chain/tests.rs]: surprised we need to ready up before running the transcript.

Comment (hdevalence) [reply zebra-consensus/src/chain/tests.rs]: for the transcript service specifically, poll_ready is a no-op, but it's required by the service contract.

File: zebra-consensus/src/chain/tests.rs
Comment (hdevalence) [reply zebra-consensus/src/chain/tests.rs:117]: Not sure about this instance but they're defined like that elsewhere so that you can have a list of transcripts and have a single test that iterates over each transcript to keep all the common code the same.

File: zebra-consensus/src/chain/tests.rs
Comment (yaahc) [reply zebra-consensus/src/chain/tests.rs]: But this is isn't the transcript service we're calling `ready` on, this is the service that we're testing with the transcript. Checking the `Transcript::check` fn shows that the first thing it does is call `ready_and`, so this looks like a violation of the service contract to me, which is why I was surprised.

Comment (hdevalence) [reply zebra-consensus/src/chain/tests.rs]: Oh that is a good point, I think you're right.

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [inline zebra-consensus/src/chain/tests.rs]: Issue: Layer returns the wrapped service, so we need to run the transcript on the return value:
https://docs.rs/tower/0.3.1/tower/layer/trait.Layer.html#tymethod.layer

```suggestion
    // Add a timeout layer
    let ready_verifier_service = TimeoutLayer::new(Duration::from_secs(VERIFY_TIMEOUT_SECONDS)).layer(&ready_verifier_service);

    let transcript = Transcript::from(BLOCK_VERIFY_TRANSCRIPT_GENESIS.iter().cloned());
    transcript.check(ready_verifier_service).await.unwrap();
```

**Review by teor2345 — DISMISSED** [review]
We need to run the transcript on the wrapped timeout service.

File: zebra-consensus/src/chain/tests.rs
Comment (teor2345) [reply zebra-consensus/src/chain/tests.rs]: I have looked at the other transcript-based tests, and they do not call ready before checking.

File: zebra-consensus/src/chain/tests.rs
Comment (oxarbitrage) [reply zebra-consensus/src/chain/tests.rs]: Applied this at https://github.com/ZcashFoundation/zebra/pull/858/commits/73ebc537a99d4b51201bcc600263c9e8b6cc905b

File: zebra-consensus/src/chain/tests.rs
Comment (oxarbitrage) [reply zebra-consensus/src/chain/tests.rs]: Removed all ready calls at https://github.com/ZcashFoundation/zebra/pull/858/commits/73ebc537a99d4b51201bcc600263c9e8b6cc905b

**Review by yaahc — APPROVED** [review]
Sweeet, tytyty

**Comment by teor2345** [comment]
I didn't get a chance to review this before it merged, so I did some fixes in PR #891.

**Pull request merged**
